src/io/config.cpp

```diff
@@ -153,6 +155,63 @@ void GetTreeLearnerType(const std::unordered_map<std::string, std::string>& para
   }
 }
 
+void Config::GetAucMuWeights(const std::unordered_map<std::string, std::string>& params) {
+  std::string filename;
+  if (Config::GetString(params, "auc_mu_weights_file", &filename)) {
+    // read weights from file
+    TextReader<size_t> reader(filename.c_str(), false);
+    Parser* parser = Parser::CreateParser(filename.c_str(), false, num_class, -1);
```

Comment:
will the `auc_mu_weight_file` be large? if not, maybe just use `ReadAllLines` and process them one-by-one...
BTW, are there any default values for the class weights when file is not provided? like ones array or random array?

Reply:
Thanks, changed as you suggested. (The weights file is small, the number of lines is equal to number of classes.) If the file is not provided, the default is equal weights for all classes (i.e. matrix is all ones with zeros on diagonal).